qcodes/instrument/parameter.py

```diff
@@ -1308,25 +1309,38 @@ def raw_value(self) -> ParamRawDataType:
             This bug will not be fixed since the `raw_value` property will be
             removed soon.
             """
-            return self._parameter.source.cache._value
+            if self._parameter.source is None:
+                return None
+            return self._parameter.source.cache.get(get_if_invalid=False)
```

Comment:
@astafan8 I am going to suggest leaving the Cache as is. The only other alternative that I can see is to make the Cache member of the parameter optional. That is a significanly more intrusive change.

Reply:
that's fine, we can get to this later